app/policies/upcoming/application_policy.rb

```diff
@@ -0,0 +1,29 @@
+class Upcoming::ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def resolve_record(record, record_type)
```

Comment:
This method allows us to pass in any of the following to do authorization checks:
```
authorize(:upcoming, :manage, @facility)
authorize(:upcoming, :mange, Facility)
authorize(:upcoming, :manage, Facility.where(...))
```
When we pass a collection or class to an authorization check, the intention is to check if the user has access to _any_ records. This is useful when we want to authorize actions for a user that aren't necessarily tied to a single resource.